File: javascript/gpt/p33.js
// p33: route that takes URL as query param and redirects if domain == example.com
const express = require("express");
const { URL } = require("url");
const app = express();

app.get("/r", (req, res) => {
  const target = req.query.url || "";
  try {
    const parsed = new URL(target);
    if (parsed.hostname === "example.com") return res.redirect(target);
  } catch (e) {}
  return res.status(400).send("Invalid URL");
});

if (require.main === module)
  app.listen(3003, () => console.log("Listening 3003"));

File: python/gpt/p33.py
# p33: Flask app route taking URL param and redirecting if domain == example.com
from flask import Flask, request, redirect
from urllib.parse import urlparse

app = Flask(__name__)


def redirect_invalid_url(url):
    parsed = urlparse(url)
    if parsed.netloc == "example.com":
        return redirect(url)
    return "Invalid URL", 400


@app.route("/r")
def r():
    target = request.args.get("url", "")
    return redirect_invalid_url(target)


if __name__ == "__main__":
    app.run(port=5002)
